Update positioning of accordion icon and support headings in accordion

bartaz opened the pull request:
## Done

Updates accordion to fix the bug with misaligned icon when text wraps.
Adds a variant to support headings in accordion buttons.

Fixes #2926 

## QA

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-3107.run.demo.haus/docs/examples/patterns/accordion/headings)
- Check accordion with headings example: https://vanilla-framework-canonical-web-and-design-pr-3107.run.demo.haus/docs/examples/patterns/accordion/headings
- Make sure documentation is fine (Codepen will show unstyled version until released): https://vanilla-framework-canonical-web-and-design-pr-3107.run.demo.haus/docs/patterns/accordion#headings


## Screenshots

<img width="504" alt="Screenshot 2020-05-29 at 15 59 27" src="https://user-images.githubusercontent.com/83575/83268286-b07f8d80-a1c5-11ea-8a91-9120860d70ce.png">
<img width="491" alt="Screenshot 2020-06-08 at 15 58 47" src="https://user-images.githubusercontent.com/83575/84039042-f6480d00-a9a0-11ea-9f92-52dd04083bfc.png">

webteam-app commented:
Starting demo at: https://vanilla-framework-canonical-web-and-design-pr-3107.run.demo.haus/

lyubomir-popov commented:
It feels like this needs a design spec - the indent becomes unnecessary when there's a difference in font sizes, and the use of utilities seems a bit too much for something that should be an out of the box component. Given the limitations imposed by the markup, I would vote for phasing it out and introducing a new pattern that supports these features out of the box, with cleaner markup, and need for utilities.

bartaz commented:
@anthonydillon @lyubomir-popov 

I updated the example as discussed in the meeting:
- added `p-accordion__title` that needs to be put on headings, this class aligns the icon with heading baseline
- added `p-accordion__tab--with-title` in place of `p-accordion__tab` - this keeps `__tab` styling, but removes the icon (because it's meant to be added by `__title`). 

Let me know what you think. We talked about having `p-accordion__button` instead, but I felt like adding `p-accordion__tab--with-title` modifier version felt better, because it doesn't introduce new class, just a modifier to existing one, keeping the backwards compatibility and familiarity.

If we want to position icon exactly we may need to have some element selectors, like here: https://github.com/canonical-web-and-design/vanilla-framework/pull/3107/files#diff-24d4aa551ccc1cb770f06ce1dab182c8R81-R90
Because we need to target specific element and adjust positioning and size of the icon.

Let me know what you think. If we are fine with this approach I'll clean up the branch and add docs.

lyubomir-popov commented on templates/docs/examples/patterns/accordion/headings.html:
```diff
@@ -0,0 +1,65 @@
+  <ul class="p-accordion__list">
+    <li class="p-accordion__group">
+      <button type="button" class="p-accordion__tab--with-title" id="tab1" role="tab" aria-controls="tab0-section" aria-expanded="true">
+        <h2 class="p-accordion__title"><span class="u-visualise-font-metrics">Advanced topics</span></h2>
```
take the utility out?

lyubomir-popov commented:
Nice work @bartaz! The only thing I notice is, the icon positioning needs a little work - currently it varies with respect to the x-height of the letters.
